sysdig/data_source_sysdig_user_test.go

```diff
@@ -36,13 +36,13 @@ func init() {
 			ctx := context.Background()
 			user, err := commonClient.GetUserByEmail(ctx, "terraform-test+user@sysdig.com")
 
-			if err != nil {
-				return err
+			if user == nil || err != nil {
+				return nil
 			}
```

Comment:
In case of error we cannot return a nil, if there's an error we need to inform the user of the provider that something wrong happened.

Reply:
I am seeing this same pattern everywhere, we need to return the error to the user of the provider, otherwise, they wont have feedback and the infrastructure will remain in an inconsistent state, because terraform will store this state but the infra is actually not modified accordingly.

Reply:
Ya I agree, I will need to change the function to not return an error when a user with that email is not found, because the user may or may not exist when this sweep() function is invoked, and it's not an error if it doesn't...

Reply:
fixed.   Also fixed getUser to pass in a query param rather than getting them all from the server and then filtering....